internal/monitors/statsd/monitor.go

```diff
@@ -0,0 +1,137 @@
+package statsd
+
+import (
+	"context"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/sfxclient"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const monitorType = "statsd"
+
+var logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 30*time.Second)
+var golibLogger = &utils.LogrusGolibShim{FieldLogger: logger.FieldLogger}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+type converterInput struct {
+	Pattern    string
+	Metric     string
+	Dimensions []string
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"false" singleInstance:"true"`
```

Comment:
There is no reason this should be `singleInstance`.  You can perfectly well have multiple statsd monitors configured in the same agent, as long as you make the ports different.